PixiEditor/Models/DataHolders/Document.cs

```diff
@@ -425,6 +471,48 @@ public void CenterContent()
                     "Center content"));
         }
 
+        private void MoveLayerProcess(object[] parameter)
+        {
+            int layerIndex = (int)parameter[0];
+            int amount = (int)parameter[1];
+
+            Layers.Move(layerIndex, layerIndex + amount);
+            if (ActiveLayerIndex == layerIndex)
+            {
+                SetActiveLayer(layerIndex + amount);
+            }
+        }
+
+        private void RestoreLayersProcess(Layer[] layers, UndoLayer[] layersData)
+        {
+            for (int i = 0; i < layers.Length; i++)
+            {
+                Layer layer = layers[i];
+
+                Layers.Insert(layersData[i].LayerIndex, layer);
+                if (layersData[i].IsActive)
+                {
+                    SetActiveLayer(Layers.IndexOf(layer));
+                }
+            }
+        }
+
+        private void RemoveLayerProcess(object[] parameters)
+        {
+            if (parameters != null && parameters.Length > 0 && parameters[0] is Guid layerGuid)
+            {
+                Layer layer = Layers.First(x => x.LayerGuid == layerGuid);
+                int index = Layers.IndexOf(layer);
+                bool wasActive = layer.IsActive;
+                Layers.Remove(layer);
+
+                if (wasActive)
+                {
+                    SetNextLayerAsActive(index);
+                }

```

Comment:
Sometimes ActiveLayerIndex in Document ends up with an incorrect value after undo/redo, causing a crash when trying to get ActiveLayer. I believe it is caused by SetNextLayerAsActive not firing when a layer that is not active gets deleted. For example, if there are two layers, and ActiveLayerIndex is 1, deleting layer 0 won't update the index, even though after the deletion there is obviously only 1 layer left and ActiveLayerIndex has to be at 0.
To replicate: make a canvas, make a new layer, delete base layer, undo, select layer 1, redo, attempt to draw.